src/frontend/screens/ObservationEdit/DescriptionField.tsx

```diff
@@ -15,36 +16,75 @@ const m = defineMessages({
 
 export const DescriptionField = () => {
   const {formatMessage: t} = useIntl();
+  const [keyboardVisible, setKeyboardVisible] = useState(false);
   const notes = usePersistedDraftObservation(store => store.value?.tags.notes);
   const {updateTags} = useDraftObservation();
 
+  useEffect(() => {
+    const keyboardHideUnsub = Keyboard.addListener('keyboardDidHide', () =>
+      setKeyboardVisible(false),
+    );
+
+    const keyboardShowUnsub = Keyboard.addListener('keyboardDidShow', () =>
+      setKeyboardVisible(true),
+    );
+
+    return () => {
+      keyboardHideUnsub.remove();
+      keyboardShowUnsub.remove();
+    };
+  }, []);
```

Comment:
This is being used in 3 places with the exact same implementation . Can you make this a custom hook?

Reply:
done